internal/unmappeddatamarshaller/unmappeddatamarshaller.go

```diff
@@ -0,0 +1,91 @@
+package unmappeddatamarshaller
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/hochfrequenz/go-bo4e/internal/jsonfieldnames"
+)
+
+type UnmappedData struct {
+	Data map[string]any
+}
+
+func (umd *UnmappedData) storeUnmappedData(key string, value any) {
+	umd.Data[key] = value
+}
+
+func MarshalWithUnmappedData[T any](s T) (bytes []byte, err error) {
+	byteArr, err := json.Marshal(s)
+	if err != nil {
+		return
+	}
+
+	unmappedDataFieldNames, err := jsonfieldnames.Extract(UnmappedData{})
+	if err != nil {
+		return
+	} else if len(unmappedDataFieldNames) != 1 {
+		err = errors.New("expected exactly one field ('Data' in struct 'UnmappedData')")
```

Comment:
Can this ever happen? It extracts the field names from `UnmappedData`, which is a concrete type. I proposed changes that would remove the type `UnmappedData` completely (see removal of `ExtensionData`), so this is probably not important.

Reply:
was removed